JustSaying/Messaging/PublishMetadata.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Collections.Generic;
+
+namespace JustSaying.Messaging
+{
+    public class PublishMetadata
+    {
+        public TimeSpan? Delay { get; set; }
+
+        public IDictionary<string, MessageAttributeValue> MessageAttributes { get; set; }
+
+        public void AddMessageAttribute(string key, MessageAttributeValue value)
+        {
+            if (MessageAttributes == null)
+            {
+                MessageAttributes = new Dictionary<string, MessageAttributeValue>();
```

Comment:
An appropriate `StringComparer` should be passed to the dictionary constructor to make the case sensitivity explicit.